web/crux/src/app/health/health.controller.ts

```diff
@@ -4,16 +4,65 @@ import { CruxHealthController, CruxHealthControllerMethods, HealthResponse } fro
 import CommonGrpcErrorInterceptor from 'src/interceptors/grpc.error.interceptor'
 import GrpcLoggerInterceptor from 'src/interceptors/grpc.logger.interceptor'
 import PrismaErrorInterceptor from 'src/interceptors/prisma-error-interceptor'
+import {
+  HealthCheckRequest,
+  HealthCheckResponse,
+  HealthCheckResponse_ServingStatus,
+  HealthController as ProtoHealthController,
+  HealthControllerMethods,
+} from 'src/grpc/protobuf/proto/health'
+import { Observable } from 'rxjs'
+import { NotFoundException, NotImplementedException } from 'src/exception/errors'
 import HealthService from './health.service'
+import HealthMapper from './health.mapper'
 
 @Controller()
 @CruxHealthControllerMethods()
+@HealthControllerMethods()
 @UseInterceptors(GrpcLoggerInterceptor, CommonGrpcErrorInterceptor, PrismaErrorInterceptor)
-export default class HealthController implements CruxHealthController {
-  constructor(private readonly service: HealthService) {}
+export default class HealthController implements CruxHealthController, ProtoHealthController {
+  constructor(private readonly service: HealthService, private mapper: HealthMapper) {}
 
   @AuditLogLevel('disabled')
   async getHealth(): Promise<HealthResponse> {
     return this.service.getHealth()
   }
+
+  @AuditLogLevel('disabled')
+  async check(request: HealthCheckRequest): Promise<HealthCheckResponse> {
+    if (!request.service || request.service === '') {
+      const health = await this.service.getHealth()
+      return {
+        status: this.mapper.serviceStatusToProto(health.status),
+      }
+    }
+    if (request.service === 'db') {
+      const health = await this.service.getHealth()
+      return {
+        status: health.lastMigration
+          ? HealthCheckResponse_ServingStatus.SERVING
+          : HealthCheckResponse_ServingStatus.NOT_SERVING,
+      }
+    }
+    if (request.service === 'kratos') {
+      const health = await this.service.getKratosHealth()
```

Comment:
You could extract the common variables like `health` outside of the if-else blocks. Minor add new lines between ifs IMHO.

Reply:
Rearranged the code a bit.